pkg/controller/controlplanecerts/controlplanecerts_controller_test.go

```diff
@@ -124,21 +114,23 @@ func TestReconcileControlPlaneCerts(t *testing.T) {
 					defaultCert("default", "secret0").
 					namedCert("cert1", "foo.com", "secret1").
 					namedCert("cert2", "bar.com", "secret2").
-					adminKubeconfig().
 					obj(),
 				fakeCertSecret("secret0"),
 				fakeCertSecret("secret1"),
 				fakeCertSecret("secret2"),
-				testAdminKubeconfigSecret(),
+			},
+			setupRemoteClient: func(mockController *gomock.Controller) *mockremoteclient.MockBuilder {
+				mockRemoteClient := mockremoteclient.NewMockBuilder(mockController)
```

Comment:
Move this code into the test proper. Use a boolean in the test case to determine whether the calls to the mock remote client builder are expected. Or use a string for the API URL to return from the mock call as a determiner of whether the calls are expected.

Reply:
i thought this approach (of allowing per-test-case remote client setup) would allow more flexibility if/when we want tests to cover error paths. although, admittedly, right now it isn't being used

Reply:
i'll change it so that the test case contains a remoteClientResponse struct to allow defining the url returned and any possible error